packages/myst-transforms/src/mathSimplifications.ts

```diff
@@ -0,0 +1,165 @@
+  if (!symbol) return undefined;
+  if (symbol.match(/^([a-zA-Z0-9+-]+)$/)) return symbol;
+  return REPLACEMENTS[symbol];
+}
+
+function replaceSymbol(node: InlineMath) {
+  const match = node.value.match(/^(\\[a-zA-Z]+)$/);
+  if (!match) return false;
+  const text = getReplacement(match[1]);
+  if (!text) return false;
+  (node as any).type = 'text';
+  node.value = text;
+  return true;
+}
+
+// function changeToEmph(node: GenericNode): Emphasis {
+//   node.type = 'emphasis';
+//   node.data = { 'specific-use': 'math' };
+//   return node as Emphasis;
+// }
+
+// function replaceSubSuperScripts(node: InlineMath, vfile: VFile) {
+//   const match = node.value.match(/^(\\?[a-zA-Z]+)(\^|_)(\\?[a-zA-Z]+)$/);
+//   if (!match) return false;
+//   const script = match[2] === '^' ? 'superscript' : 'subscript';
+//   const first = getReplacement(match[1]);
+//   const second = getReplacement(match[3]);
+//   if (!first || !second) return false;
+//   const emph = changeToEmph(node);
+//   emph.children = [
+//     { type: 'text', value: first },
+//     { type: script, children: [{ type: 'text', value: second }] },
+//   ] as any;
+//   return true;
+// }
+
+// function replaceTextCommand(node: InlineMath, vfile: VFile) {
+//   const match = node.value.match(/^\\text\{([a-zA-Z0-9-\s]+)}$/);
+//   if (!match) return false;
+//   const text = match[1];
+//   if (!text) return false;
+//   (node as any).type = 'text';
+//   node.value = text;
+//   return true;
+// }
+
+function replaceDirectSubSuperScripts(node: InlineMath) {
+  const match = node.value.match(/^(\^|_)(?:(?:\{(\\?[a-zA-Z0-9+-]+)\})|(\\?[a-zA-Z0-9+-]+))$/);
+  if (!match) return false;
+  const script = match[1] === '^' ? 'superscript' : 'subscript';
+  const value = getReplacement(match[2] || match[3]);
+  if (!value) return false;
+  if (value === '∘' && script === 'superscript') {
+    (node as any).type = 'text';
+    node.value = '°';
+    return true;
+  }
+  (node as any).type = script;
+  (node as any).children = [{ type: 'text', value }];
+  delete (node as any).value;
+  return true;
+}
+
+function replaceNumberedSubSuperScripts(node: InlineMath) {
+  const match = node.value.match(/^([+-]?[\d.]+)(\^|_)(?:(?:\{([+-]?[\d.]+)\})|([+-]?[\d.]+))$/);
+  if (!match) return false;
+  const first = match[1];
+  const second = match[3] || match[4];
+  const script = match[2] === '^' ? 'superscript' : 'subscript';
+  (node as any).type = 'span';
+  (node as any).children = [
+    { type: 'text', value: first },
+    { type: script, children: [{ type: 'text', value: second }] },
+  ];
+  delete (node as any).value;
+  return true;
+}
+
+function replaceNumber(node: InlineMath) {
+  const match = node.value.match(/^(-?[0-9.]+)$/);
+  if (!match) return false;
+  const text = match[1];
+  if (!text) return false;
+  (node as any).type = 'text';
+  node.value = text;
+  return true;
+}
+
+export function inlineMathSimplificationTransform(mdast: GenericParent) {
```

Comment:
Ah, so this totally eliminates `inlineMath` nodes if they can be simply represented by symbols and basic markdown. If the inline math has any marginal complexity at all (interleaved sub/superscript, macros, unknown symbols, etc), it is just left alone and handled by whatever latex math renderer.

Reply:
Yeah, I think we can have some settings on this in the future. Most (I think all) of these are also captured on the way back to latex, however will be `\textsuperscript{th}` instead of `$^{th}`, which is actually correct!